### PR TITLE
Move from Maven to Gradle

### PR DESCRIPTION
**Pull request by danielpassos**
This PR is includes:

* [FH-3156](https://issues.jboss.org/browse/FH-3156) - Migrate fh-android-sdk build from Maven to Gradle 
* [FH-3347](https://issues.jboss.org/browse/FH-3347) - Update fh-android-sdk minSdkVersion to 16 
* [FH-3348](https://issues.jboss.org/browse/FH-3348) - Update fh-android-sdk test to the new Android testing framework 
* [FH-3350](https://issues.jboss.org/browse/FH-3350) - Add public test cover using coveralls.io
* [FH-3362](https://issues.jboss.org/browse/FH-3362) -  Add some shields bageds on fh-android-sdk README 
* [FH-3363](https://issues.jboss.org/browse/FH-3363) - Update fh-android-sdk README to follow the same format of AeroGear project

### CODE REVIEW

**Comment by danielpassos** [comment]
@secondsun I really want your review here, especially of README and test. Double check if I didn't forgot to add `@Test` on some method :pray:

File: README.md [hunk 1]
```diff
@@ -1,75 +1,85 @@
-# FeedHenry Android SDK [![Build Status](https://travis-ci.org/feedhenry/fh-android-sdk.png)](https://travis-ci.org/feedhenry/fh-android-sdk)
+# FeedHenry Android SDK
 
-[![Maven Central](https://maven-badges.herokuapp.com/maven-central/com.feedhenry/fh-android-sdk/badge.svg?style=flat-square)](https://maven-badges.herokuapp.com/maven-central/com.feedhenry/fh-android-sdk/)
+[![Travis](https://img.shields.io/travis/feedhenry/fh-android-sdk.svg)](http://travis-ci.org/feedhenry/fh-android-sdk)
+[![Coveralls](https://img.shields.io/coveralls/feedhenry/fh-android-sdk/master.svg)](https://coveralls.io/github/feedhenry/fh-android-sdk)
+[![License](https://img.shields.io/badge/-Apache%202.0-blue.svg)](https://opensource.org/s/Apache-2.0)
```
Comment (jcesarmobile) [inline README.md]: The link is broken, should be https://opensource.org/licenses/Apache-2.0 or http://www.apache.org/licenses/LICENSE-2.0

File: README.md [hunk 2]
```diff
@@ -1,75 +1,85 @@
...
 [![Javadocs](http://www.javadoc.io/badge/com.feedhenry/fh-android-sdk.svg?color=blue)](http://www.javadoc.io/doc/com.feedhenry/fh-android-sdk)
 
+This SDK should provide you with all you'll need to start developing cloud-connected apps with the FeedHenry platform. The SDK provides access to cloud action calls, app authentication and authorization.   
 
-This SDK should provide you with all you'll need to start developing cloud-connected apps with the FeedHenry platform. The SDK provides access to cloud action calls, app authentication and authorization. 
+|                 | Project Info                                                 |
+| --------------- | ------------------------------------------------------------ |
+| License:        | Apache License, Version 2.0                                  |
+| Build:          | Gradle                                                       |
+| Documentation:  | http://feedhenry.org/                                        |
```
Comment (jcesarmobile) [inline README.md]: http://feedhenry.org/docs/ has a link to the customer portal, I could only find this which isn't up to date. https://access.redhat.com/documentation/en-us/red_hat_mobile_application_platform/4.2/html/client_sdk/native-android

Not sure if we have a better one, but I don't think we should use http://feedhenry.org/

File: README.md [hunk 3]
```diff
@@ -1,75 +1,85 @@
...
-### Gradle Projects (Recommended)
-
-Add the FH Android `aar` library in your `build.gradle` project
-
-```
-dependencies {
-	compile 'com.feedhenry:fh-android-sdk:3.1.0'
-}
-```
+For more details about that please consult [FH Android SDK Guide](http://docs.feedhenry.com/v3/dev_tools/sdks/android.html).
```
Comment (jcesarmobile) [inline README.md]: This link redirects to https://access.redhat.com/documentation/en/red-hat-mobile-application-platform-hosted/, so as I commented before, maybe use this direct link to Android SDK instead  https://access.redhat.com/documentation/en-us/red_hat_mobile_application_platform/4.2/html/client_sdk/native-android

**Comment by danielpassos** [comment]
> @danielpassos once you add the packaging instructions in I will test with some template apps.  In general I am +1 though.

Going to create a new PR/issue for that.

**Comment by danielpassos** [comment]
> Also, deps/android-async-http-1.4.9.jar and deps/httpclient-4.3.6.jar were removed, but only android-async-http is in library/build.gradle, is httpclient not needed anymore?

@jcesarmobile httpclient is a dependency in [android-async-http](https://github.com/loopj/android-async-http/blob/master/library/build.gradle#L28)

> Also, the whole doc folder was removed, has it been moved elsewhere? or it isn't needed anymore?

@jcesarmobile Yes, It's automatically publish in http://javadoc.io/ for every release. There is badge linking to [there](http://www.javadoc.io/doc/com.feedhenry/fh-android-sdk)

File: README.md (hunk 2 above)
Comment (danielpassos) [inline README.md]: @jcesarmobile I got your point BUT. Feedhenry SDK is an Open Source library and should point to the Feedhenry website. This link is about RHMAP (product) :/

File: README.md (hunk 3 above)
Comment (danielpassos) [inline README.md]: Same here!

File: README.md (hunk 2 above)
Comment (danielpassos) [inline README.md]: @johnfriz @lgriffin Wdyt?

**Comment by danielpassos** [comment]
@jcesarmobile Apart of use `https://access.redhat.com/documentation/` for docs. Done!

File: README.md (hunk 2 above)
Comment (jcesarmobile) [inline README.md]: The thing is, feedhenry website has no documentation for the SDK, so makes no sense to link it. The docs section only links the customer portal. What about using the javadoc link instead? http://www.javadoc.io/doc/com.feedhenry/fh-android-sdk

**Comment by danielpassos** [comment]
@jcesarmobile Done!

**Pull request merged**